cypress/support/import_test_util.js

```diff
@@ -85,19 +85,24 @@ function setUpSavingStubs() {
 
 /**
  * A wrapper for {@link convertEeObjectToPromise} that returns a resolve
- * function for releasing the result.
- * @return {Function}
+ * function for releasing the result and a promise that resolves when the
+ * convert starts. This allows us to make multiple calls to this method
+ * and wait on the start promises, therefore ensuring that an instance of
+ * oldConvert never points to the new promise-waiting function.
+ * @return {{resolveFunction: Function, startPromise: Promise<void>}}
  */
 function getConvertEeObjectToPromiseRelease() {
-  let resolveFunction = null;
-  const promise = new Promise((resolve) => resolveFunction = resolve);
+  let releaseLatch = null;
+  const convertFinishLatch = new Promise((resolve) => releaseLatch = resolve);
+  let startFunction = null;
+  const convertStartPromise = new Promise((resolve) => startFunction = resolve);
   const oldConvert = MapUtil.convertEeObjectToPromise;
   MapUtil.convertEeObjectToPromise = (eeObject) => {
     MapUtil.convertEeObjectToPromise = oldConvert;
-    return MapUtil.convertEeObjectToPromise(eeObject).then(async (result) => {
-      await promise;
-      return result;
-    });
+    startFunction();
+    return Promise
+        .all([MapUtil.convertEeObjectToPromise(eeObject), convertFinishLatch])
+        .then((results) => results[0]);
   };
-  return resolveFunction;
+  return {startPromise: convertStartPromise, releaseLatch: releaseLatch};
```

Comment:
nit: can omit `releaseLatch: `. And if you named "convertStartPromise" to "startPromise", could omit that as well.